crates/cli/src/subcommands/start.rs

```diff
@@ -52,13 +58,16 @@ pub async fn exec(paths: &SpacetimePaths, args: &ArgMatches) -> anyhow::Result<E
         .arg(&paths.cli_config_dir)
         .args(args);
 
-    // TODO(noa): use std::os::unix::process::CommandExt::exec() here once we have windows CI
-    // use std::os::unix::process::CommandExt;
-    // let err = cmd.exec();
-    // Err(err).context(format!("exec failed for {}", bin_path.display()))
+    #[cfg(unix)]
+    let result = {
+        use std::os::unix::process::CommandExt;
+        // if exec() succeeds, it diverges, so the function just returns an io::Error
+        let err = cmd.exec();
+        Err::<std::process::ExitStatus, _>(err)
+    };
+    #[cfg(not(unix))]
+    let result = cmd.status();
 
-    let status = cmd
-        .status()
-        .with_context(|| format!("exec failed for {}", bin_path.display()))?;
+    let status = result.with_context(|| format!("exec failed for {}", bin_path.display()))?;
```

Comment:
I believe these changes are to address https://github.com/clockworklabs/SpacetimeDB/issues/2133. 

Given that that issue has a different purpose and surface area for testing than the update functionality, I would like to suggest addressing it in a separate PR.

Reply:
Split out into #2200.